haskell/src/Monocle/Version.hs

```diff
@@ -0,0 +1,14 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Monocle version
+module Monocle.Version (version) where
+
+import Development.GitRev (gitDirty, gitHash)
+import Relude
+
+version :: String
+version = $(gitHash) <> dirty
```

Comment:
To display the version from the cabal file, use:

```suggestion
import Data.Version (showVersion)
import Paths_monocle (version)

version :: String
version = concat [showVersion version, " ", $(gitHash), dirty]
```

And don't forget to add in the cabal file:

```
  other-modules:    Paths_monocle
  autogen-modules:  Paths_monocle
```

Reply:
What is Paths_monocle module ?

Reply:
it is something cabal generate for us, to access the info from the cabal file.